src/main/resources/templates/fragments/login-fragment.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:th="http://www.thymeleaf.org" xmlns:sec="http://www.thymeleaf.org/thymeleaf-extras-springsecurity4">
+<head>
+    <meta charset="UTF-8"/>
+
+</head>
+<body>
+<div th:fragment="login-modal">
+    <div id="login" class="modal fade">
+       <div class="modal-content">
+          <div class="modal-header">
+              <button type="button" class="close" data-dismiss="modal">&times;</button>
+              <h4 class="modal-title">Please Log In</h4>
+          </div>
+           <div class="modal-body">
+             <div th:if="${param.error}">
```

Comment:
What `param`?

Reply:
`param` referes to the values in the query string. It doesn't need to be defined anywhere. It'll be available in any page.